lib/backends/sdc/containers.js

```diff
@@ -3578,6 +3578,14 @@ function _runExec(opts, callback) {
 
     serverSocket.on('connect', setupListeners);
 
+    // error can happen before connect too (eg. ECONNREFUSED)
+    serverSocket.on('error', function (error) {
+        opts.log.error('serverSocket for %s threw an error %s',
+            cmdString, error.toString());
+
+        endSocket(error);
```

Comment:
Why do we use `endSocket` here and not in `_runAttach`? It seems that `endSocket` would display a message such as `'client socket threw an error`, but here we're dealing with a socket that is connected to the CN. So is `serverSocket.end()` here more appropriate?